packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L2DepositedERC20.sol

```diff
@@ -193,7 +193,27 @@ contract OVM_L2DepositedERC20 is iOVM_L2DepositedToken, OVM_CrossDomainEnabled,
         virtual
         onlyFromCrossDomainAccount(address(l1TokenGateway))
     {
-        // todo verify _l1Token matches
+        // Verify the deposited token on L1 matches the L2 deposited token representation here
+        // Otherwise immediately queue a withdrawal
+        if(_l1Token != address(l1Token)) {
+
+            bytes memory message = abi.encodeWithSelector(
+                iOVM_L1TokenGateway.finalizeWithdrawal.selector,
+                _l1Token,
+                address(this),
+                _from,
+                _to,
+                _amount,
+                _data
+            );
+
+            // Send message up to L1 gateway
+            sendCrossDomainMessage(
+                address(l1TokenGateway),
+                0,
```

Comment:
This looks a bit odd. What if instead of setting 0 gas in multiple places, we had a second function on OVM_CrossDomainEnabled, which could have the same or a different name from this one.

```
function sendCrossDomainMessage(
        address _crossDomainTarget,
        bytes memory _message
    )
        internal
    {
        getCrossDomainMessenger().sendMessage(_crossDomainTarget, 0, _gasLimit);
    }
```

Reply:
We discussed, and decided not too as the L2 Messenger will eventually handle this argument.